### PR TITLE
Added unit testing documentation

### PR DESCRIPTION
**Pull request by sammyshj**
@smcgregor Changes made in the PR:
Added unit testing documentation.
To do: If the current changes look good to you, then, the integration testing steps will be added to the same PR too.

### CODE REVIEW

**Comment by smcgregor** [comment]
Looks good.

**Comment by sammyshj** [comment]
@smcgregor Added both unit testing and integration testing documentation. Let me know if any further changes are required. Thanks.

File: docs/testing.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Integration Testing
+-------------------
+The extension presently does not support integration testing with
+the help of `Selenium <http://www.seleniumhq.org>`_. So, to make
+sure that the extension is functioning as expected, we need to
+manually follow the steps given below to ensure the same.
+
+To test injected link functionality:
+
+- Navigate to :code:`http://test.privly.org/test_pages/url_parameters.html` on the browser.
```
Comment (smcgregor) [inline docs/testing.rst]: Add a few URLs you will visit to ensure they still inject properly, like the "host page tests" on test.privly.org.

File: docs/testing.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+- The :code:`No Parameters` link will get replaced with an injected iframe.
+- This injected iframe contains the privly message.
+- Click on the privly popover and select :code:`Stop Injecting`.
+- The injected iframe will now get replaced with the original text.
+- Repeat the test by navigating to :code:`https://www.facebook.com/mohammed.smith.7` on the browser.
```
Comment (sammyshj) [inline docs/testing.rst]: @smcgregor Added the host page iframe injection test. Presently, the extension does not work on Twitter due to CSP and hence the twitter links are not mentioned right now.

Comment (smcgregor) [inline docs/testing.rst]: I would still add it as a negative test case. We can't ship this to users until we close #34, and when it is closed we don't want a regression.

Comment (sammyshj) [inline docs/testing.rst]: Added a new test, `To test injected link functionality with host page CSP` as a negative test case.

**Pull request merged**